services/distribution/src/main/java/app/coronawarn/server/services/distribution/objectstore/S3RetentionPolicy.java

```diff
@@ -53,69 +59,110 @@ public S3RetentionPolicy(ObjectStoreAccess objectStoreAccess, DistributionServic
    */
-  public void applyRetentionPolicy(int retentionDays) {
+  public void applyDiagnosisKeyDayRetentionPolicy(int retentionDays) {
     Set<String> countries = Set.of(originCountry, euPackageName);
     countries.forEach(country -> {
-      List<S3Object> diagnosisKeysObjects = objectStoreAccess.getObjectsWithPrefix(getPrefix(country));
+      List<S3Object> diagnosisKeysObjects = objectStoreAccess.getObjectsWithPrefix(getDiagnosisKeyPrefix(country));
       final LocalDate cutOffDate = TimeUtils.getUtcDate().minusDays(retentionDays);
       diagnosisKeysObjects.stream()
-          .filter(diagnosisKeysObject -> isFilePathOlderThan(diagnosisKeysObject, cutOffDate))
-          .forEach(this::deleteDiagnosisKey);
+          .filter(diagnosisKeysObject -> isDiagnosisKeyFilePathOlderThan(diagnosisKeysObject, cutOffDate))
+          .forEach(this::deleteS3Object);
     });
   }
 
   /**
-   * Delete old hourly files based on retention policy.
+   * Deletes all diagnosis-key hour files from S3 that are older than retentionDays.
    *
    * @param retentionDays number of days back where hourly files should be deleted
    */
-  public void applyHourFileRetentionPolicy(long retentionDays) {
+  public void applyDiagnosisKeyHourRetentionPolicy(long retentionDays) {
     Set<String> countries = Set.of(originCountry, euPackageName);
     countries.forEach(country -> {
-      List<S3Object> diagnosisKeysObjects = objectStoreAccess.getObjectsWithPrefix(getPrefix(country));
+      List<S3Object> diagnosisKeysObjects = objectStoreAccess.getObjectsWithPrefix(getDiagnosisKeyPrefix(country));
       final LocalDate cutOffDate = TimeUtils.getUtcDate().minusDays(retentionDays - 1L);
       var deletableKeys = diagnosisKeysObjects.stream()
-          .filter(diagnosisKeysObject -> isFilePathOlderThan(diagnosisKeysObject, cutOffDate))
-          .filter(this::isFilePathOnHourFolder)
+          .filter(diagnosisKeysObject -> isDiagnosisKeyFilePathOlderThan(diagnosisKeysObject, cutOffDate))
+          .filter(this::isDiagnosisKeyFilePathOnHourFolder)
           .collect(Collectors.toList());
 
-      logger.info("Deleting {} files from hourly folders older than {}", deletableKeys.size(), cutOffDate.toString());
-      deletableKeys.forEach(this::deleteDiagnosisKey);
+      logger.info("Deleting {} diagnosis key files from hourly folders older than {}", deletableKeys.size(),
+          cutOffDate.toString());
+      deletableKeys.forEach(this::deleteS3Object);
+    });
+  }
+
+  /**
+   * Deletes all trace time warning hour files from S3 that are older than retentionDays.
+   *
+   * @param retentionDays number of days back where hourly files should be deleted
+   */
+  public void applyTraceTimeWarningHourRetentionPolicy(long retentionDays) {
+    Set<String> countries = Set.of(originCountry, euPackageName);
+    countries.forEach(country -> {
+      List<S3Object> traceTimeWarningsObjects = objectStoreAccess
```

Comment:
where do we host the TraceTimeWarningPackages? also under the country paths?

Reply:
Yes, but with a different prefix "twp"